### ISSUE TITLE
no `CsrfToken` extracted if path has info

### ISSUE THREAD

**Issue by Conni2461** [edward-shen/actix-csrf#7]
So yeah if we have a path info like `/user/{id}` we cant get the CsrfToken. The issue for this is that currently its checked if `req.path()` (`/user/1`) is in a HashMap but obv the map only contains `/user/{id}`. I currently can't think about a great solution other than first doing a contains and then iterating over it and checking if there is a match. (we could check what actix-web is doing for these paths)

Example (just in case i am not clear):

```rs
use actix_csrf::extractor::CsrfToken;
use actix_csrf::CsrfMiddleware;
use actix_web::http::Method;
use actix_web::HttpResponse;
use actix_web::{web, App, HttpServer, Responder};
use rand::prelude::StdRng;

#[actix_web::main]
async fn main() -> std::io::Result<()> {
    HttpServer::new(|| {
        let csrf = CsrfMiddleware::<StdRng>::new().set_cookie(Method::GET, "/user/{id}");

        App::new()
            .wrap(csrf)
            .route("/user/{id}", web::get().to(user_ui))
    })
    .bind(("127.0.0.1", 8080))?
    .run()
    .await
}

async fn user_ui(token: CsrfToken, _path: web::Path<u32>) -> impl Responder {
    // ERROR
    let body = format!(
        r#"
        <!DOCTYPE html>
        <html>
        <head><meta charset="UTF-8"><title>Example</title></head>
        <body>
        <form action="/login" method="post">
            <input type="hidden" name="csrf_token" value="{}" />
            <button type="submit">Submit</button>
        </form>
        </body>
        </html>
        "#,
        token.get()
    );

    HttpResponse::Ok().body(body)
}
```

So yeah opening `/user/1` will now yield a 422.

None the less, thanks for the crate everything else went pretty smoothly (my `into_inner` PR ignored) :)

**Comment by Conni2461** [comment]
Nvm i know a good solution, I will post a PR: https://github.com/edward-shen/actix-csrf/pull/8

**Comment by edward-shen** [comment]
Merged with #8.
